### PR TITLE
Publish Data API jar to maven central

### PR DESCRIPTION
**Pull request by kathirsvn**
**What this PR does**:
Changes to publish Data API jar to maven central

**Checklist**
- [X] Changes manually tested
- [X] Automated Tests added/updated
- [X] Documentation added/updated
- [X] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by kathirsvn** [comment]
This PR must be merged only after merging https://github.com/stargate/data-api/pull/1330 and rebasing the branch of this PR with the main, since the PR above removes offline mode code which must be completed before publishing this jar to the maven central

File: pom.xml [hunk 1]
```diff
@@ -435,5 +435,89 @@
...
+                <goals>
+                  <goal>jar-no-fork</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <version>3.4.1</version>
```
Comment (tatu-at-datastax) [inline pom.xml:464]: nit: if we have to specify version (it doesn't come from parent pom), latest is `3.8.0`.

File: pom.xml [hunk 2]
```diff
@@ -435,5 +435,89 @@
...
+                  <placement>a</placement>
+                  <head>API note:</head>
+                </tag>
+              </tags>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-gpg-plugin</artifactId>
+            <version>3.0.1</version>
```
Comment (tatu-at-datastax) [inline pom.xml:489]: ... and latest here is `3.2.4`

**Review by tatu-at-datastax — APPROVED** [review]

**Comment by jeffreyscarpenter** [comment]
Closing, decided to keep publishing to private repo. See #1337.

**Pull request closed without merging**